bitcoin/src/pow.rs

```diff
@@ -248,6 +249,20 @@ impl Target {
     /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
     /// adjustment period.
     pub fn max_difficulty_transition_threshold(&self) -> Self { Self(self.0 << 2) }
+
+    /// Returns the least number of bits needed to represent the number.
+    pub fn bits(&self) -> u32 { self.0.bits() }
+
+    /// Wrapping multiplication by `u64`.
+    ///
+    /// # Returns
+    ///
+    /// The multiplication result along with a boolean indicating whether an arithmetic overflow
+    /// occurred. If an overflow occurred then the wrapped value is returned.
+    pub fn mul_u64(self, rhs: u64) -> (Target, bool) {
```

Comment:
This should be private, right? Isn't the whole motivation for adding the `calculate_next_work_required` function so we don't expose arbitrary math operations on the target/work types?

Reply:
Correct